vscode/vscode/vscode.mli

```diff
@@ -1718,23 +1718,15 @@ module TaskProvider : sig
   val t_to_js : t -> Ojs.t
 
   val provideTasks :
-    t -> ?token:CancellationToken.t -> unit -> Task.t list ProviderResult.t
+    t -> token:CancellationToken.t -> Task.t list ProviderResult.t
 
   val resolveTasks :
-       t
-    -> task:Task.t
-    -> ?token:CancellationToken.t
-    -> unit
-    -> Task.t ProviderResult.t
+    t -> task:Task.t -> token:CancellationToken.t -> Task.t ProviderResult.t
 
   val create :
-       provideTasks:
-         (?token:CancellationToken.t -> unit -> Task.t list ProviderResult.t)
+       provideTasks:(token:CancellationToken.t -> Task.t list ProviderResult.t)
```

Comment:
Not entirely related to this PR, but do we need the `token` label here?

Reply:
Not necessarily, but I'd like to keep the labels consistent across the bindings.

Reply:
Agreed with Rudi: since we have type `CancellationToken.t`, don't really need label but up to you